cellfinder/napari/curation.py

```diff
@@ -373,16 +384,45 @@ def save_training_data(
         prompt_for_directory :
             If `True` show a file dialog for the user to select a directory.
         """
+
         if self.is_data_extractable():
             if prompt_for_directory:
                 self.get_output_directory()
+                # if the directory is not empty
+                if any(self.output_directory.iterdir()):
+                    choice = display_warning(
+                        self,
+                        "About to save training data",
+                        "Existing files will be will be deleted. Proceed?",
+                    )
+                    if not choice:
+                        return
             if self.output_directory is not None:
+                self.__prep_directories_for_save()
                 self.__extract_cubes(block=block)
                 self.__save_yaml_file()
                 show_info("Done")
 
             self.update_status_label("Ready")
 
+    def __prep_directories_for_save(self):
+        self.yaml_filename = self.output_directory / "training.yml"
+        self.cell_cube_dir = self.output_directory / "cells"
+        self.no_cell_cube_dir = self.output_directory / "non_cells"
+
+        self.__delete_existing_saved_training_data()
+
+    def __delete_existing_saved_training_data(self):
```

Comment:
For my understanding, why do functions in the file start with two underscores??

Reply:
Double underscores [are as close to private as you can get in Python](https://docs.python.org/3/tutorial/classes.html#private-variables). Not sure why they're used here and not anywhere else in BrainGlobe though.

Reply:
Ah cool! Didn't know. From reading that it sounds like the use case is mainly to make it impossible to override this in subclasses. We currently don't have any subclasses of `CurationWidget`, and for consistency it would be good to not use it here, but then again not sure it's enough of a priority right now :thinking: 

TLDR I suggest leaving for now?

Reply:
Fine by me.